flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/inference/CallBindingCallContext.java

```diff
@@ -62,6 +65,7 @@ public final class CallBindingCallContext extends AbstractSqlCallContext {
     private final List<DataType> argumentDataTypes;
     private final @Nullable DataType outputType;
     private final @Nullable List<StaticArgument> staticArguments;
+    private final SqlValidator validator;
```

Comment:
validator does not really fit in here, can we avoid it? SqlModelCall should have resolved types already. I think TableArgCall has the same? we should synchronize the two if possible.